templates/js-langchain/src/vector_index_cache.js

```diff
@@ -0,0 +1,71 @@
+import { Actor } from 'apify';
+import { createHash } from 'crypto';
+import { pipeline } from 'node:stream/promises';
+import tar from 'tar';
+
+const VECTOR_INDEX_CACHE_STORE_NAME = 'vector-index-cache';
+
+const client = Actor.newClient();
+
+/**
+ * Generates vector index cache key as a hash of provided configuration object.
+ *
+ * @param {Object} config
+ * @returns {String}
+ */
+function getIndexCacheKey(config) {
+    const hash = createHash('md5').update(JSON.stringify(config)).digest('hex');
+
+    return `${hash}.tar.gz`;
+}
+
+/**
+ * Returns an ID of a key-value store with a name VECTOR_INDEX_CACHE_STORE_NAME.
+ *
+ * @returns {String}
+ */
+async function getVectorIndexCacheStore() {
+    const { id: vectorIndexCacheStoreId } = await client.keyValueStores().getOrCreate(VECTOR_INDEX_CACHE_STORE_NAME);
+
+    return client.keyValueStore(vectorIndexCacheStoreId);
```

Comment:
Why not just `Actor.openKeyValueStore(VECTOR_INDEX_CACHE_STORE_NAME)`? Or if you want to have it on the platform, use `{ forceCloud: true )`, but I think it's better to save it locally anyway.